prov/gni/include/gnix.h

```diff
@@ -339,19 +340,26 @@ struct gnix_fab_req {
 	void                      *user_context;
 	/* matched_rcv_fab_req only applicable to GNIX_FAB_RQ_RECV type */
 	struct gnix_fab_req       *matched_rcv_fab_req;
-	void     *buf;
 	/* current point in the buffer for next transfer chunk -
 	   case of long messages or rdma requests greater than 4 GB */
 	void     *cur_pos;
-	size_t   len;
 	uint64_t imm;
-	uint64_t tag;
 	struct gnix_vc *vc;
 	void *completer_data;
-	int (*completer_func)(void *,int *);
+	int (*completer_func)(void *);
 	int modes;
 	int retries;
 	uint32_t id;
+
+	/* RMA stuff */
+	uint64_t loc_addr;
+	void *loc_md;
+	uint64_t rem_addr;
+	uint64_t rem_mr_key;
+	size_t len;
+	uint64_t flags;
+	uint64_t data;
+	uint64_t tag;
```

Comment:
Not for this commit, but eventually we'll probably need to figure out how to compact this structure.  Its getting pretty big.